src/main/java/io/netty/incubator/codec/http3/QpackDecoderStateSyncStrategy.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.incubator.codec.http3;
+
+/**
+ * A strategy that determines when to send <a
+ * href="https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#section-2.2.2.3">acknowledgment</a> of new table
+ * entries on the QPACK decoder stream.
+ */
+public interface QpackDecoderStateSyncStrategy {
+
+    /**
+     * Callback when an <a
+     * href="https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#name-encoded-field-section-prefi">
+     * encoded header field section</a> is decoded successfully by the decoder.
+     *
+     * @param requiredInsertCount for the encoded field section.
+     */
+    void sectionAcknowledged(int requiredInsertCount);
+
+    /**
+     * When a header field entry is added to the decoder dynamic table.
+     *
+     * @param insertCount for the entry.
+     * @return {@code true} if an <a
+     * href="https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#name-insert-count-increment">insert count
+     * increment decoder instruction</a> should be sent.
+     */
+    boolean entryAdded(long insertCount);
+
+    /**
+     * Returns a {@link QpackDecoderStateSyncStrategy} that will acknowledge each entry added via
+     * {@link #entryAdded(long)}.
+     * @return A {@link QpackDecoderStateSyncStrategy} that will acknowledge each entry added via
+     * {@link #entryAdded(long)}.
+     */
+    static QpackDecoderStateSyncStrategy ackEachInsert() {
+        return new QpackDecoderStateSyncStrategy() {
```

Comment:
nit: store this instance somewhere and return the same on each call. This implementation is stateless so no need to create a new one everytime.

Reply:
So I figured out this class needs state 😄